code/modules/reagents/chemistry/items.dm

```diff
@@ -0,0 +1,162 @@
+	if(!istype(cont))
+		return
+	if(used == TRUE)
+		to_chat(user, "<span class='warning'>[src] has already been used!</span>")
+		return
+	if(!LAZYLEN(cont.reagents.reagent_list))
+		return
+	switch(round(cont.reagents.pH, 1))
+		if(14 to INFINITY)
+			color = "#462c83"
+		if(13 to 14)
+			color = "#63459b"
+		if(12 to 13)
+			color = "#5a51a2"
+		if(11 to 12)
+			color = "#3853a4"
+		if(10 to 11)
+			color = "#3f93cf"
+		if(9 to 10)
+			color = "#0bb9b7"
+		if(8 to 9)
+			color = "#23b36e"
+		if(7 to 8)
+			color = "#3aa651"
+		if(6 to 7)
+			color = "#4cb849"
+		if(5 to 6)
+			color = "#b5d335"
+		if(4 to 5)
+			color = "#f7ec1e"
+		if(3 to 4)
+			color = "#fbc314"
+		if(2 to 3)
+			color = "#f26724"
+		if(1 to 2)
+			color = "#ef1d26"
+		if(-INFINITY to 1)
+			color = "#c6040c"
+	desc += " The paper looks to be around a pH of [round(cont.reagents.pH, 1)]"
+	name = "used [name]"
+	used = TRUE
+
+/*
+* pH meter that will give a detailed or truncated analysis of all the reagents in of an object with a reagents datum attached to it. Only way of detecting purity for now.
+*/
+/obj/item/pHmeter
+	name = "Chemistry Analyser"
+	desc = "A a electrode attached to a small circuit box that will tell you the pH of a solution. The screen currently displays nothing."
+	icon_state = "pHmeter"
+	icon = 'icons/obj/chemical.dmi'
+	resistance_flags = FLAMMABLE
+	w_class = WEIGHT_CLASS_TINY
+	///level of detail for output for the meter
+	var/scanmode = DETAILED_CHEM_OUTPUT 
+
+/obj/item/pHmeter/attack_self(mob/user)
+	if(!scanmode)
+		to_chat(user, "<span class='notice'>You switch the chemical analyzer to give a detailed report.</span>")
+		scanmode = DETAILED_CHEM_OUTPUT
+	else
+		to_chat(user, "<span class='notice'>You switch the chemical analyzer to give a reduced report.</span>")
+		scanmode = SHORTENED_CHEM_OUTPUT
+
+/obj/item/pHmeter/afterattack(atom/A, mob/user, proximity)
+	. = ..()
+	if(!istype(A, /obj/item/reagent_containers))
+		return
+	var/obj/item/reagent_containers/cont = A
+	if(LAZYLEN(cont.reagents.reagent_list) == null)
+		return
+	var/out_message
+	to_chat(user, "<i>The chemistry meter beeps and displays:</i>")
+	out_message += "<span class='notice'><b>Total volume: [round(cont.volume, 0.01)] Total pH: [round(cont.reagents.pH, 0.01)]\n"
+	out_message += "Chemicals found in the beaker:</b>\n"
+	if(cont.reagents.is_reacting)
+		out_message += "<span class='warning'>A reaction appears to be occuring currently.<span class='notice'>\n"
+	for(var/datum/reagent/R in cont.reagents.reagent_list)
+		out_message += "<b>[round(R.volume, 0.01)]u of [R.name]</b>, <b>Purity:</b> [round(R.purity, 0.01)], [(scanmode?"[(R.overdose_threshold?"<b>Overdose:</b> [R.overdose_threshold]u, ":"")][(R.addiction_threshold?"<b>Addiction:</b> [R.addiction_threshold]u, ":"")]<b>Base pH:</b> [R.pH].":"<b>Base pH:</b> [R.pH].")]\n"
+		if(scanmode)
+			out_message += "<b>Analysis:</b> [R.description]\n"
+	to_chat(user, "[out_message]</span>")
```

Comment:
```suggestion
	var/list/out_message = list()
	to_chat(user, "<i>The chemistry meter beeps and displays:</i>")
	out_message += "<span class='notice'><b>Total volume: [round(cont.volume, 0.01)] Total pH: [round(cont.reagents.pH, 0.01)]\n"
	out_message += "Chemicals found in the beaker:</b>\n"
	if(cont.reagents.is_reacting)
		out_message += "<span class='warning'>A reaction appears to be occuring currently.<span class='notice'>\n"
	for(var/datum/reagent/R in cont.reagents.reagent_list)
		out_message += "<b>[round(R.volume, 0.01)]u of [R.name]</b>, <b>Purity:</b> [round(R.purity, 0.01)], [(scanmode?"[(R.overdose_threshold?"<b>Overdose:</b> [R.overdose_threshold]u, ":"")][(R.addiction_threshold?"<b>Addiction:</b> [R.addiction_threshold]u, ":"")]<b>Base pH:</b> [R.pH].":"<b>Base pH:</b> [R.pH].")]\n"
		if(scanmode)
			out_message += "<b>Analysis:</b> [R.description]\n"
	to_chat(user, "[out_message.Join()]</span>")
```
BYOND is awfully bad at handling strings, but has a better time with lists.
Use this pattern for compiling bits of text. You can also add the `"\n"` glue to the `Join()` proc and you'll reduce code a little bit if you want.